src/rpc/mining.cpp

```diff
@@ -354,16 +354,183 @@ std::string gbt_vb_name(const Consensus::DeploymentPos pos)
     return s;
 }
 
+// Sets the version bits in a block
+static void UtilMkBlockTmplVersionBits(std::set<std::string> setClientRules,
+    CBlockIndex *pindexPrev,
+    int64_t coinbaseSize,
+    CBlock *pblock,
+    UniValue *paRules,
+    UniValue *pvbavailable)
+{
+    const Consensus::Params &consensusParams = Params().GetConsensus();
```

Comment:
Wonder if we could call this just once in `mkblocktemplate()` and pass it as parameter to `UtilMkBlockTmplVersionBits` and `MkFullMiningCandidateJson`.